### PR TITLE
Add flags to allow tuning k8s client throttling.

### PR DESCRIPTION
**Pull request by jkh52**
I have seen that if enough agents exist, and one of a set of masters is unavailable (so every agent is sync-ing), proxy server goroutines grows unbounded. With pprof I find that client-go throttling (of token reviews) is where they are blocked.

In my repro, if I replace the client-go defaults (5 qps, 10 burst) with higher numbers, it mitigates the issue.

### CODE REVIEW

File: cmd/server/main.go [hunk 1]
```diff
@@ -121,6 +121,10 @@ type ProxyRunOptions struct {
 	authenticationAudience string
 	// Path to kubeconfig (used by kubernetes client)
 	kubeconfigPath string
+	// Client maximum QPS.
+	kubeconfigQps float32
```
Comment (Jefftree) [inline cmd/server/main.go:125]: Stylistically capitalizing QPS is probably more correct.

**Review by Jefftree — COMMENTED** [review]
One nit comment, otherwise LGTM.

File: cmd/server/main.go [hunk 2]
```diff
@@ -375,6 +385,14 @@ func (p *Proxy) run(o *ProxyRunOptions) error {
 			return fmt.Errorf("failed to load kubernetes client config: %v", err)
 		}
 
+		if o.kubeconfigQps != 0 {
+			klog.V(1).Infoln("Setting k8s client QPS: %v", o.kubeconfigQps)
```
Comment (Jefftree) [inline cmd/server/main.go:389]: Also from the build:

```
# sigs.k8s.io/apiserver-network-proxy/cmd/server
cmd/server/main.go:389:4: Infoln call has possible formatting directive %v
cmd/server/main.go:393:4: Infoln call has possible formatting directive %v
```

**Comment by jkh52** [comment]
Thanks.

File: cmd/server/main.go [hunk 3]
```diff
@@ -154,6 +158,8 @@ func (o *ProxyRunOptions) Flags() *pflag.FlagSet {
 	flags.StringVar(&o.agentNamespace, "agent-namespace", o.agentNamespace, "Expected agent's namespace during agent authentication (used with agent-service-account, authentication-audience, kubeconfig).")
 	flags.StringVar(&o.agentServiceAccount, "agent-service-account", o.agentServiceAccount, "Expected agent's service account during agent authentication (used with agent-namespace, authentication-audience, kubeconfig).")
 	flags.StringVar(&o.kubeconfigPath, "kubeconfig", o.kubeconfigPath, "absolute path to the kubeconfig file (used with agent-namespace, agent-service-account, authentication-audience).")
+	flags.Float32Var(&o.kubeconfigQPS, "kubeconfig-qps", o.kubeconfigQPS, "Maximum client QPS (used with authentication).")
```
Comment (caesarxuchao) [inline cmd/server/main.go:161]: nit: there are multiple authentications in the system. Perhaps "proxy server uses this client to authenticate agent tokens" to be more specific.

**Comment by caesarxuchao** [comment]
A nit. LGTM otherwise.

File: cmd/server/main.go (hunk 3 above)
Comment (jkh52) [reply cmd/server/main.go:161]: Done.

**Pull request merged**